selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/platform/administration/ContentAdministration.java

```diff
@@ -911,13 +911,27 @@ public void deleteDocumentTemplate(String title) {
    * @param content
    */
   public void addTemplateInList(String name, String nameTemplate, String content) {
-    evt.click(ELEMENT_ECM_TEMPLATES_LIST_ADD_LIST);
-    evt.type(ELEMENT_ECM_TEMPLATES_LIST_TEMPLATE_NAME_FORM, nameTemplate, true);
-    evt.type(ELEMENT_ECM_TEMPLATES_LIST_NAME_FORM, name, true);
-    evt.type(ELEMENT_ECM_TEMPLATES_LIST_CONTENT_FORM, content, true);
-    evt.click(ELEMENT_ECM_TEMPLATES_LIST_SAVE_FORM);
-    evt.waitForAndGetElement(By.xpath(ELEMENT_ECM_TEMPLATES_LIST_CHECK_LIST.replace("{$name}", name).replace("{$template}",
-                                                                                                             nameTemplate)));
+    $(ELEMENT_ECM_TEMPLATES_LIST_ADD_LIST).click();
+    $(ELEMENT_ECM_TEMPLATES_LIST_TEMPLATE_NAME_FORM).setValue(nameTemplate);
+    $(ELEMENT_ECM_TEMPLATES_LIST_NAME_FORM).setValue(name);
+    $(ELEMENT_ECM_TEMPLATES_LIST_CONTENT_FORM).setValue(content);
+    $(ELEMENT_ECM_TEMPLATES_LIST_SAVE_FORM).click();
+    $(ELEMENT_ECM_TEMPLATES_LIST_SAVE_FORM).waitUntil(Condition.not(Condition.visible),Configuration.timeout);
+      for (int i = 0; i < 9; i++) {
+        if ($(byXpath(ELEMENT_ECM_TEMPLATES_LIST_CHECK_LIST.replace("{$name}", name).replace("{$template}",
+                nameTemplate)))
+                .is(Condition.not(Condition.visible))) {
+          $(ELEMENT_ICON_NEXT_ARROW).click();
+        }
+        if ($(byXpath(ELEMENT_ECM_TEMPLATES_LIST_CHECK_LIST.replace("{$name}", name).replace("{$template}",
+                nameTemplate)))
+                .waitUntil(Condition.visible,Configuration.timeout).isDisplayed()) {
+          break;
+        }
+      }
```

Comment:
What is the purpose of this loop ? Why 10 iterations ?
Also, the indentation is not good.

Reply:
the purpose of this loop is to check if the template is added, if the template is not found on the current page,  click on next page, until finding the template

Reply:
For the indentation, code formatted .

Reply:
> the purpose of this loop is to check if the template is added, if the template is not found on the current page, click on next page, until finding the template

OK, so please use a constant for the value with a meaningful name, and add a comment to explain that.

Reply:
done (y)